rr_platform/src/bigoli_motor_relay_node/bigoli_ethernet_drive_relay.cpp

```diff
@@ -0,0 +1,170 @@
+#include <ros/ros.h>
+#include <rr_platform/speed.h>
+#include <rr_platform/steering.h>
+#include <rr_platform/chassis_state.h>
+#include <boost/asio.hpp>
+#include <boost/array.hpp>
+#include <iostream>
+
+/*@NOTE THIS CODE USES BOOST 1.58 as it is the version currently installed with ROS.
+  If that changes, this code will need to be updated as such!
+  So don't fear if it breaks, just fix the things by checking the links below to examples given.
+*/
+
+using namespace std;
+using boost::asio::ip::tcp;
+
+struct PIDConst {
+  float p;
+  float i;
+  float d;
+};
+
+struct PIDConst accelDrivePID, decelDrivePID, steeringPID;
+
+double speed = 0.0;
+double steeringAngle = 0.0;
+
+double maxAngleMsg;
+const double maxOutput = 1.0;  //#TODO: magic # to launch param maybe???
+
+std::shared_ptr<tcp::socket> currentSocket;
```

Comment:
I'm glad we used smart pointers here. Kind of nitpicking, but best practice is to use unique_ptr and make_unique for this use case. We only want to keep one reference to it (it is not a shared resource between multiple contexts)